src/integrations/stripe.js

```diff
@@ -0,0 +1,101 @@
+import { loadStripe } from '@stripe/stripe-js';
+
+const stripePromise = new Promise((resolve) => {
+  document.addEventListener('stripe_key_loaded', (e) => {
+    const promise = loadStripe(e.detail.key);
+    promise.then((stripe) => {
+      resolve(stripe);
+    });
+  });
+});
+
+export function getStripePromise() {
+  return stripePromise;
+}
+export async function getStripe() {
```

Comment:
the async and await can be removed here, and you can just call await getStripe()

Reply:
export const getStripe = () => getStripePromise()

but this also seems redundant with the above function

Reply:
Yes, simplification is in order here.  Combining them into one.